pkg/cluster/deploy.go

```diff
@@ -30,3 +47,92 @@ func (u *simpleUpgrader) Deploy(ctx context.Context, cs *api.OpenShiftManagedClu
 	}
 	return nil
 }
+
+func hashScaleSets(azuretemplate map[string]interface{}) (map[string]string, error) {
+	ssHashes := make(map[string]string)
+	for _, r := range jsonpath.MustCompile("$.resources[?(@.type='Microsoft.Compute/virtualMachineScaleSets')]").Get(azuretemplate) {
+		resource, ok := r.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		// cleanup capacity so that no unnecessary VM rotations are going
+		// to occur because of a scale up/down.
+		jsonpath.MustCompile("$.sku.capacity").Delete(resource)
+
+		// hash scale set
+		data, err := json.Marshal(resource)
+		if err != nil {
+			return nil, err
+		}
+		hf := sha256.New()
+		hf.Write(data)
+
+		scaleSetName := jsonpath.MustCompile("$.name").MustGetString(resource)
+		// drop the ss- prefix
+		ssHashes[scaleSetName[3:]] = base64.StdEncoding.EncodeToString(hf.Sum(nil))
```

Comment:
`hex.EncodeToString` is probably preferable to `base64.StdEncoding.EncodeToString` for a hash?

Reply:
We use the latter in sync. No strong feelings.